Sources/SwiftCli/main.swift

```diff
@@ -12,12 +12,14 @@ struct StartProgram: ParsableCommand{
     @Flag(help: "Set if you want your pattern to be knit flat. Default value is knit in the round.")
     var knitFlat = false
 
-
     func run() {
-        let inputValidator = InputValidator()
+        let patternNormalizer = PatternNormalizer()
+        let nestedArrayBuilder = NestedArrayBuilder()
+        let inputValidator = InputValidator(patternNormalizer: patternNormalizer, nestedArrayBuilder: nestedArrayBuilder)
```

Comment:
Some follow up to our last pairing discussion on configuring the dependencies to inject -- I have no problem with what you've done here.  It's easy for me to see what's going on and is some good use of dependency injection.  Nice job.